### ISSUE TITLE
Language Simplification: Unified Syntax for Definitions, Sessions, and Annotations

### ISSUE THREAD

**Issue by arthur-debert** [arthur-debert/txxt-rust#141]
# Language Simplification: Unified Annotation Syntax

Implements the proposal from `docs/proposals/simplifying.txxt` - unifying metadata handling around annotations and using whitespace rules to distinguish between sessions and definitions.

### Annotations (✅ Phase 1 Complete)
- **Old**: `:: label : params ::`
- **New**: `:: label params ::` (whitespace separator)
- **New**: `:: params ::` (params-only, no label)

### Definitions (❌ Not Yet Implemented)
- **Old**: `Term ::`
- **New**: `Term:` (single colon, immediate indented content, no blank line)
- **Metadata**: Optional trailing annotation `:: def.label key=value ::`

### Sessions
- **Pattern**: Title + **mandatory blank line** + indented content
- **Distinguisher**: Blank line presence (vs definition's immediate content)
- **Context**: Only allowed at DocumentRoot or inside other Sessions

### Verbatim
- **Pattern**: Like definitions but with **mandatory** trailing annotation
- **Example**: 
  ```
  Code example:
      console.log("hello");
  :: javascript edition=es6 ::
  ```

### ✅ Completed (PR #140)
- Phase 1.1: Line classification functions (`is_definition_marker()`, `is_blank_line()`)
- Phase 1.2: Annotation parameter syntax change (whitespace separator)
- Phase 1.3: Context-aware detection functions (`is_definition_start()`, `is_session_start()`, `is_paragraph_line()`)

### ❌ Remaining Work
- **Definition Parsing**: Update `transform_definition()` to use new `Term:` syntax instead of old `Term ::` syntax
- **Integration**: The parsing code in `transform_complex_pattern()` still checks for `TxxtMarker` (::) instead of `Colon` (:)
- **Testing**: Update test corpus and unit tests for new definition syntax

## Why This Matters

1. **Parser Simplification**: Session vs Definition distinction becomes a simple blank line check
2. **Unified Metadata**: Annotations become the single way to handle parameters and labels
3. **Reduced Cognitive Load**: One core pattern instead of multiple special cases
4. **Natural Grouping**: Trailing annotations can apply to multiple blocks

## Implementation Focus

The minimal work to complete simplification:
1. Change `transform_complex_pattern()` line 914-926 to look for `Colon` instead of `TxxtMarker`
2. Update `transform_definition()` to parse `Term:` syntax
3. Update test corpus and unit tests
4. Verify end-to-end flow works

**Note**: The context-aware functions (`is_definition_start()`, etc.) already exist but are not yet integrated into the main parser. Full integration is optional - the minimal changes above are sufficient to make the new syntax work.

## Related

- Proposal: `docs/proposals/simplifying.txxt`
- Completed work: PR #140
- Code location: `src/syntax/semantic_analysis.rs` (lines 914-927, 1649-1683)
